src/cache.ts

```diff
@@ -6,22 +7,31 @@ export interface ObjectCache<T> {
     list(namespace?: string): ReadonlyArray<T>;
 }
 
-export type ListCallback<T extends KubernetesObject> = (list: T[]) => void;
-
-export class ListWatch<T extends KubernetesObject> implements ObjectCache<T> {
+export class ListWatch<T extends KubernetesObject> implements ObjectCache<T>, Informer<T> {
     private objects: T[] = [];
     private readonly indexCache: { [key: string]: T[] } = {};
+    private readonly callbackCache: { [key: string]: Array<ObjectCallback<T>> } = {};
 
     public constructor(
         private readonly path: string,
         private readonly watch: Watch,
-        private readonly listFn: (callback: ListCallback<T>) => void,
+        private readonly listFn: ListPromise<T>,
     ) {
         this.watch = watch;
         this.listFn = listFn;
         this.doneHandler(null);
     }
 
+    public on(verb: string, cb: ObjectCallback<T>) {
```

Comment:
Not sure this should be a blocker but we can't actually remove handlers?

Reply:
I was waffling on this too.  If you're going to start adding more event handling / management methods, consider re-using the [EventEmitter](https://nodejs.org/api/events.html) (*e.g.*, extending it).  You'll need to define your own generic interface for it, however.

You could also just wait until someone requests functionality the `EventEmitter` implements (like removing).

Reply:
I think maybe we should get this out and get feedback on how useful this is. I feel like this is overkill for a first attempt and we are using the correct names so we could switch it out later.